packages/nox/src-loca/src-thyra/LOCA_Thyra_Group.C

```diff
@@ -480,25 +480,35 @@ LOCA::Thyra::Group::setSaveDataStrategy(
 }
 
 void
-LOCA::Thyra::Group::updateThyraParamView()
+LOCA::Thyra::Group::updateThyraParam()
 {
   if (paramsInSeparatePVecs) {
+    // Create thyra vector to store LOCA parameters
     param_thyra_vec.resize(param_index.size());
     for (size_t i=0; i < param_thyra_vec.size(); ++i) {
-      const RTOpPack::ConstSubVectorView<double> pv(Teuchos::ArrayRCP<const double>(params.getDoubleArrayPointer()+i,0,1,false));
-      Teuchos::RCP<const ::Thyra::VectorSpaceBase<double> > ps =
-        model_->get_p_space(param_index[i]);
-      param_thyra_vec[i] = ::Thyra::createMemberView(ps,pv);
+      Teuchos::RCP<const ::Thyra::VectorSpaceBase<double> > ps = model_->get_p_space(param_index[i]);
+      if(ps->dim() != 1)
+        globalData->locaErrorCheck->throwError(
+           "LOCA::Thyra::Group::updateThyraParam()",
+           "Thyra parameter space should have dimension 1. Each scalar parameter is stored in a separate vector");
+      const Teuchos::RCP< ::Thyra::VectorBase<double> > param_thyra_vec_setup = ::Thyra::createMember(ps);
+      ::Thyra::put_scalar(params.getValue(i), param_thyra_vec_setup.ptr());
+      param_thyra_vec[i] = param_thyra_vec_setup;
```

Comment:
I think this is a new restriction that will break charon. A parameter vector can be a block of parameters and Charon sometimes uses this for a block of boundary constraints. Maybe instead of this change, we should fix the Thyra::createMemberView for tpetra objects. During ATDM we changed over all of what we needed at the time, but did not fix all of thyra.

Reply:
> I think this is a new restriction that will break charon. A parameter vector can be a block of parameters and Charon sometimes uses this for a block of boundary constraints. Maybe instead of this change, we should fix the Thyra::createMemberView for tpetra objects. During ATDM we changed over all of what we needed at the time, but did not fix all of thyra.

Albany uses parameter blocks as well, but it is using the conditional body after this (for `paramsInSeparatePVecs == false`). I assume Charon does the same. If my understanding is correct, I'm not adding a new restriction, just throwing an error for clarity.

Reply:
> > I think this is a new restriction that will break charon. A parameter vector can be a block of parameters and Charon sometimes uses this for a block of boundary constraints. Maybe instead of this change, we should fix the Thyra::createMemberView for tpetra objects. During ATDM we changed over all of what we needed at the time, but did not fix all of thyra.
> 
> Albany uses parameter blocks as well, but it is using the conditional body after this (for `paramsInSeparatePVecs == false`). I assume Charon does the same. If my understanding is correct, I'm not adding a new restriction, just throwing an error for clarity.

I should add that I tested the code with Albany and it works fine.